Makefile

```diff
@@ -1,4 +1,17 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+ifndef BUILD_DIRECTORY
```

Comment:
So by using `$(dir $(BUILD_DIRECTORY)) == ./` we ensure its not an absolute path, which doesn't mean anything, since the directory name might or might not exist, and there can still be any level of depth (`foo/bar/baz/release`).

What about changing only the rule that builds `$(BUILD_DIRECTORY)` to `mkdir -p`?